src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/BusinessEventValidator.java

```diff
@@ -50,6 +54,74 @@ public boolean validate(
     return validateMandatoryEvents(eventMap) && validateNonMandatoryEvents(eventMap);
   }
 
+  /** Validates the dates on event creation */
+  public boolean validateOnCreate(
+      final List<Event> existingEvents,
+      final Event newEvent,
+      final CollectionExerciseDTO.CollectionExerciseState collectionExerciseState) {
+    Map<String, Event> events =
+        existingEvents.stream().collect(Collectors.toMap(Event::getTag, Function.identity()));
+    if (collectionExerciseState.equals(CollectionExerciseDTO.CollectionExerciseState.CREATED)) {
+      return validateMandatoryEventsOnCreate(events, newEvent);
+    }
+    return false;
+  }
+
+  /** Validates the mandatory dates on event creation. Event dates can be added in any order. */
+  private boolean validateMandatoryEventsOnCreate(
+      final Map<String, Event> eventMap, Event newEvent) {
+    final Optional<Event> mpsEvent =
+        Optional.ofNullable(eventMap.get(EventService.Tag.mps.toString()));
+    final Optional<Event> goLiveEvent =
+        Optional.ofNullable(eventMap.get(EventService.Tag.go_live.toString()));
+    final Optional<Event> returnByEvent =
+        Optional.ofNullable(eventMap.get(EventService.Tag.return_by.toString()));
+    final Optional<Event> exerciseEndEvent =
+        Optional.ofNullable(eventMap.get(EventService.Tag.exercise_end.toString()));
+
+    List<Event> events = new ArrayList<>();
+    if (newEvent.getTag().equals(EventService.Tag.mps.toString())) {
+      events.add(newEvent);
+    } else if (mpsEvent.isPresent()) {
```

Comment:
If this is only used on event create should the `.isPresent` not be the first check? Or is the behaviour allowing updates at this point as well?

Reply:
Yes, the idea is to also use this for updates.